pinot-common/src/main/java/com/linkedin/pinot/common/data/StarTreeIndexSpec.java

```diff
@@ -16,66 +16,78 @@
 package com.linkedin.pinot.common.data;
 
 import com.google.common.base.Objects;
-
-import java.util.Collections;
 import java.util.List;
 import java.util.Set;
 
 public class StarTreeIndexSpec {
-  public static final Integer DEFAULT_MAX_LEAF_RECORDS = 100000; // TODO: determine a good number via experiment
+  public static final int DEFAULT_MAX_LEAF_RECORDS = 100000; // TODO: determine a good number via experiment
   public static final int DEFAULT_SKIP_MATERIALIZATION_CARDINALITY_THRESHOLD = 10000;
+  public static final int DEFAULT_SKIP_SPLIT_ON_TIME_COLUMN_THRESHOLD = 1000;
+
+  /** The upper bound on the number of leaf records to be scanned for any query. */
+  private int _maxLeafRecords = DEFAULT_MAX_LEAF_RECORDS;
+
+  /** The threshold to decide whether to skip or split on columns if not explicitly specified. */
+  private int _skipMaterializationCardinalityThreshold = DEFAULT_SKIP_MATERIALIZATION_CARDINALITY_THRESHOLD;
 
-  /** The upper bound on the number of leaf records to be scanned for any query */
-  private Integer maxLeafRecords = DEFAULT_MAX_LEAF_RECORDS;
+  /** The threshold to decide whether to further split on time column for leaf nodes. */
+  private int _skipSplitOnTimeColumnThreshold = DEFAULT_SKIP_SPLIT_ON_TIME_COLUMN_THRESHOLD;
 
-  /** Dimension split order (if null or absent, descending w.r.t. dimension cardinality) */
-  private List<String> dimensionsSplitOrder;
+  /** Dimension split order (if null or absent, descending w.r.t. dimension cardinality). */
+  private List<String> _dimensionsSplitOrder;
```

Comment:
I think this is a very good way to distinguish member variables and local variables, and inside IDE it's showing java style warning on these names (forgive my OCD :P). We can keep the old convention, but since it's private, no functionality will change for this class.